### ISSUE TITLE
[explicit-function-return-type] Decorators syntax

### ISSUE THREAD

**Issue by valerio-battaglia** [typescript-eslint/typescript-eslint#505]
**What code were you trying to parse?**

```ts
import { ObjectType, Field } from 'type-graphql'

@ObjectType()
export class Test {
  @Field(type => String)
  public name: string = ''
}

export default Test
```

**What did you expect to happen?**

Neither errors nor warnings.

**What actually happened?**

`5:10 Missing return type on function. eslint(@typescript-eslint/explicit-function-return-type)`

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `^1.7.0` |
| `@typescript-eslint/parser` | `^1.7.0` |
| `TypeScript`                           | `^3.4.5` |
| `node`                                 | `10.15.3` |
| `npm`                                  | `6.4.1` |

**Comment by bradzacher** [comment]
It looks like a valid error.

Your decorator contains an arrow function which has no return type:

```ts
import { ObjectType, Field } from 'type-graphql'

@ObjectType()
export class Test {
  @Field(type => String) //////// has no return type
  public name: string = ''
}

export default Test
```

**Comment by valerio-battaglia** [comment]
If I correct the code as below, TS complains:
```
import { ObjectType, Field } from 'type-graphql'

@ObjectType()
export class Test {
  @Field((returns): ParentInfo[] => [ParentInfo]) //////// TS complain about the return which is not incorrect
  public name: string = ''
}

export default Test
```

**Comment by valerio-battaglia** [comment]
It works!
